vote.py

```diff
@@ -0,0 +1,172 @@
+from brutal.core.plugin import BotPlugin, cmd
+
+
+class Vote(BotPlugin):
+    def setup(self, *args, **kwargs):
+        self.reset()
+
+    def reset(self):
+        """Reset all voting variables"""
+        self.voting = False
+        self.voting_q = False
+        self.yeses = 0
+        self.nos = 0
+        self.voted = {}
+
+    def judgment(self, event=None, penalty=''):
+        """Sumarize votes and make decision"""
+        # TODO: if voting succeeded, get op and punish user
+        msg = 'unsuccessfully'
+        if self.yeses + self.nos > 1 and self.yeses > self.nos:
+            msg = 'successfully '
+
+        msg += 'with {0}x yes, {1}x no'.format(self.yeses, self.nos)
+        punishment = '{0} {1}'.format(penalty, event.args[0])
+
+        self.reset()
+
+        return 'Voting to {0} finished {1}'.format(punishment, msg)
+
+    def vote_punishment(self, event=None, penalty=''):
+        """Start voting of given penalty"""
+        if self.voting or self.voting_q:
+            return 'Voting in progress...'
+
+        if not len(event.args):
+            return 'No user specified'
+
+        user = event.args[0]
+        self.voting = True
+
+        self.delay_task(60, self.judgment, event=event, penalty=penalty)
+
+        return 'Voting to {0} {1} started for 60 secs. (!yes / !no)' \
+            .format(penalty, user)
+
+    # TODO: put all these commands into one regexp match
+    @cmd
+    def votekick(self, event):
+        return self.vote_punishment(event, 'kick')
+
+    @cmd
+    def voteban(self, event):
+        return self.vote_punishment(event, 'ban')
+
+    @cmd
+    def voteunban(self, event):
+        return self.vote_punishment(event, 'unban')
+
+    @cmd
+    def votemute(self, event):
+        return self.vote_punishment(event, 'mute')
+
+    def yesnoquestion_result(self, event=None):
+        """Return results of yes/no question"""
+        yes = []
+        no = []
+        for k, v in self.voted.iteritems():
+            if v[1]:
+                yes.append(v[0])
+            elif not v[1]:
+                no.append(v[0])
+
+        res = '{0}x yes ({2}), {1}x no ({3})'.format(self.yeses,
```

Comment:
Even if there are zero nos for instance? If I read this correctly, this will return something like 

```
2x yes (adman -> fero, madman -> taraba), 0x no ()